homeassistant/components/unifiprotect/media_player.py

```diff
@@ -0,0 +1,139 @@
+"""Support for Ubiquiti's UniFi Protect NVR."""
+from __future__ import annotations
+
+from collections.abc import Callable, Sequence
+import logging
+from typing import Any
+
+from pyunifiprotect.data import Camera
+from pyunifiprotect.exceptions import StreamError
+
+from homeassistant.components.media_player import (
+    DEVICE_CLASS_SPEAKER,
+    MediaPlayerEntity,
+    MediaPlayerEntityDescription,
+)
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MUSIC,
+    SUPPORT_PLAY_MEDIA,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_STOP,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_IDLE, STATE_PLAYING
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .data import ProtectData
+from .entity import ProtectDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[Sequence[Entity]], None],
+) -> None:
+    """Discover cameras with speakers on a UniFi Protect NVR."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            ProtectMediaPlayer(
+                data,
+                camera,
+            )
+            for camera in data.api.bootstrap.cameras.values()
+            if camera.feature_flags.has_speaker
+        ]
+    )
+
+
+class ProtectMediaPlayer(ProtectDeviceEntity, MediaPlayerEntity):
+    """A Ubiquiti UniFi Protect Speaker."""
+
+    def __init__(
+        self,
+        data: ProtectData,
+        camera: Camera,
+    ) -> None:
+        """Initialize an UniFi speaker."""
+
+        self.device: Camera = camera
+        self.entity_description = MediaPlayerEntityDescription(
+            key="speaker", device_class=DEVICE_CLASS_SPEAKER
+        )
+        super().__init__(data)
+
+        self._attr_name = f"{self.device.name} Speaker"
+        self._attr_supported_features = (
+            SUPPORT_PLAY_MEDIA
+            | SUPPORT_VOLUME_SET
+            | SUPPORT_VOLUME_STEP
+            | SUPPORT_STOP
+            | SUPPORT_SELECT_SOURCE
```

Comment:
We don't offer a way to select source?

Reply:
The speakers on the cameras are pretty limited. It was mostly designed to just let you play TTS messages. They can only play MP3 file URIs that Home Assistant can access.

If you have a reference to an integration with similar requiresments that lets you select a MP3 file, I can add the browse for it.

Reply:
If we don't support selecting source we should remove this support.

Reply:
Oh, you mean there. I think that is needed for TTS to appear? At least I think that is why I added it, I would need to verify.

Reply:
We don't add supported features that we don't support, regardless of other things.

Reply:
Removing SELECT_SOURCE does not affect TTS. So, I have it removed in my next PR.